avs_core/filters/turn.cpp

```diff
@@ -498,11 +579,18 @@ static void turn_180_yuy2(const BYTE* srcp, BYTE* dstp, int src_rowsize, int src
 
 Turn::Turn(PClip c, int direction, IScriptEnvironment* env) : GenericVideoFilter(c), u_source(nullptr), v_source(nullptr)
 {
-    num_planes = (vi.pixel_type & VideoInfo::CS_INTERLEAVED) ? 1 : 3;
+    if (vi.pixel_type & VideoInfo::CS_INTERLEAVED) {
+        num_planes = 1;
+    } else if (vi.IsPlanarRGBA() || vi.IsYUVA()) {
+        num_planes = 4;
+    } else {
+        num_planes = 3;
+    }
 
     splanes[0] = 0;
-    splanes[1] = PLANAR_U;
-    splanes[2] = PLANAR_V;
+    splanes[1] = vi.IsYUV() ? PLANAR_U : PLANAR_B;

```

Comment:
At the moment IsYUV and IsYUVA are mutually exclusive by definition in avisynth.h
It was an early decision, I don't know if it is right anyway.
The reason was to avoid existing filters incorrectly assuming that they handle a 3-plane color space while YUVA has four planes. Same logic applies to IsPlanarRGB and IsPlanarRGBA

I check it this way in ConvertTo16bit::GetFrame:
  if(vi.IsPlanar())
  {
      int planes_y[4] = { PLANAR_Y, PLANAR_U, PLANAR_V, PLANAR_A };
      int planes_r[4] = { PLANAR_G, PLANAR_B, PLANAR_R, PLANAR_A };
      int *planes = (vi.IsYUV() || vi.IsYUVA()) ? planes_y : planes_r;
      for (int p = 0; p < vi.NumComponents(); ++p) {
          const int plane = planes[p];
          conv_function(src->GetReadPtr(plane), dst->GetWritePtr(plane),
              src->GetRowSize(plane), src->GetHeight(plane),
              src->GetPitch(plane), dst->GetPitch(plane), float_range /*, dither_mode */);
      }
  }